soc/ti/simplelink/cc23x0/ccfg.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 Texas Instruments Incorporated
+ * Copyright (c) 2024 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include <inc/hw_ccfg.h>
+
+#define CC23_CCFG_FLASH            DT_INST(0, ti_cc23x0_ccfg_flash)
+#define CC23_CCFG_FLASH_PROP(prop) DT_PROP(CC23_CCFG_FLASH, prop)
+
+#define CC23_TO_PERM_VAL(en) ((en) ? CCFG_PERMISSION_ALLOW : CCFG_PERMISSION_FORBID)
+
+#if CONFIG_CC23X0_BLDR_VTOR_TYPE_UNDEF
+#define CC23X0_BLDR_VTOR 0xffffffff
+#elif CONFIG_CC23X0_BLDR_VTOR_TYPE_FORBID
+#define CC23X0_BLDR_VTOR 0xfffffffc
+#elif CONFIG_CC23X0_BLDR_VTOR_TYPE_USE_FCFG
+#define CC23X0_BLDR_VTOR 0xfffffff0
+#else
+#define CC23X0_BLDR_VTOR CONFIG_CC23X0_BLDR_VTOR_FLASH
+#endif
+
+#if CC23_CCFG_FLASH_PROP(ti_chip_erase) == 0
+#warning ti,chip-erase property is NOT PRESENT in your device tree, \
+	 flashing this firmware will LOCK YOUR DEVICE.
+#endif
+
+/* Default CCFG */
+const ccfg_t ccfg __attribute__((section(".ti_ccfg"))) __attribute__((used)) = {
+
+	.bootCfg.pBldrVtor = (void *)CC23X0_BLDR_VTOR,
+
+	.bootCfg.bldrParam.serialRomBldrParamStruct.bldrEnabled =
+		IS_ENABLED(CONFIG_CC23X0_BLDR_ENABLED),
+	.bootCfg.bldrParam.serialRomBldrParamStruct.serialIoCfgIndex =
+		CONFIG_CC23X0_SERIAL_IO_CFG_INDEX,
+	.bootCfg.bldrParam.serialRomBldrParamStruct.pinTriggerEnabled =
+		IS_ENABLED(CONFIG_CC23X0_PIN_TRIGGER_ENABLED),
+	.bootCfg.bldrParam.serialRomBldrParamStruct.pinTriggerDio = CONFIG_CC23X0_PIN_TRIGGER_DIO,
+	.bootCfg.bldrParam.serialRomBldrParamStruct.pinTriggerLevel =
+		IS_ENABLED(CONFIG_CC23X0_PIN_TRIGGER_LEVEL_HI),
+	.bootCfg.pAppVtor = (void *)CONFIG_CC23X0_P_APP_VTOR,
+
+	.hwOpts = {CC23_CCFG_FLASH_PROP(ti_hw_opts_1), CC23_CCFG_FLASH_PROP(ti_hw_opts_2)},
+
+	.permissions.allowDebugPort = CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_debug_port)),
+	.permissions.allowEnergyTrace = CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_energy_trace)),
+	.permissions.allowFlashVerify = CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_flash_verify)),
+	.permissions.allowFlashProgram = CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_flash_program)),
+	.permissions.allowChipErase = CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_chip_erase)),
+	.permissions.allowToolsClientMode =
+		CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_tools_client_mode)),
+	.permissions.allowReturnToFactory =
+		CC23_TO_PERM_VAL(CC23_CCFG_FLASH_PROP(ti_ret_to_factory)),
+	.permissions.allowFakeStby = CCFG_PERMISSION_ALLOW,
+
+	.misc.saciTimeoutOverride = CC23_CCFG_FLASH_PROP(ti_saci_timeout_override),
```

Comment:
saciTimeoutOverride should no longer be configurable, and should be removed from devicetree. This struct-member should be hardcoded to 0